nexus/src/db/collection_insert.rs

```diff
@@ -112,12 +112,65 @@ pub trait DatastoreCollection<ResourceType> {
             <Self::CollectionTimeDeletedColumn as Column>::Table,
         ): TypesAreSame,
         Self: Sized,
-        ResourceTable<ResourceType, Self>: Copy + Debug,
+        // Enables the "table()" method.
+        CollectionTable<ResourceType, Self>: HasTable<Table = CollectionTable<ResourceType, Self>>
+            + 'static
+            + Send
+            + Table
+            // Allows calling ".into_boxed()" on the table.
+            + query_methods::BoxedDsl<
+                'static,
+                Pg,
+                Output = BoxedDslOutput<CollectionTable<ResourceType, Self>>,
+            >,
+        // Allows treating "filter_subquery" as a boxed "dyn QueryFragment<Pg>".
+        <CollectionTable<ResourceType, Self> as QuerySource>::FromClause:
+            QueryFragment<Pg> + Send,
+        // Allows sending "filter_subquery" between threads.
+        <CollectionTable<ResourceType, Self> as AsQuery>::SqlType: Send,
+        // Allows calling ".filter()" on the boxed table.
+        BoxedQuery<CollectionTable<ResourceType, Self>>:
+            query_methods::FilterDsl<
+                    Eq<
+                        CollectionPrimaryKey<ResourceType, Self>,
+                        CollectionId<ResourceType, Self>,
+                    >,
+                    Output = BoxedQuery<CollectionTable<ResourceType, Self>>,
+                > + query_methods::FilterDsl<
+                    IsNull<CollectionTimeDeletedColumn<ResourceType, Self>>,
+                    Output = BoxedQuery<CollectionTable<ResourceType, Self>>,
+                >,
+        // Allows using "key" in in ".eq(...)".
+        CollectionId<ResourceType, Self>: diesel::expression::AsExpression<
+            SerializedCollectionPrimaryKey<ResourceType, Self>,
+        >,
+        <CollectionPrimaryKey<ResourceType, Self> as Expression>::SqlType:
+            SingleValue,
+        // Allows calling "is_null()" on the time deleted column.
+        CollectionTimeDeletedColumn<ResourceType, Self>: ExpressionMethods,
+        // Necessary for output type (`InsertIntoCollectionStatement`).
```

Comment:
I appreciate the comments here on what each bit does. This is wild.

Reply:
Ah I see, these were mostly moved, not added. Still the comments are a good addition. I'm sure you had to figure all that out from scratch.

Reply:
Yeah, I tried going through each bound, disabling it, and providing context based on compiler errors. Hopefully documentation can be the first step to simplification.